contracts/strategies/VariableRewardsStrategy.sol

```diff
@@ -0,0 +1,304 @@
+        if (devFee > 0) {
+            rewardToken.safeTransfer(devAddr, devFee);
+        }
+
+        uint256 reinvestFee = amount.mul(REINVEST_REWARD_BIPS).div(BIPS_DIVISOR);
+        if (reinvestFee > 0) {
+            rewardToken.safeTransfer(msg.sender, reinvestFee);
+        }
+
+        uint256 depositTokenAmount = _convertRewardTokenToDepositToken(amount.sub(devFee).sub(reinvestFee));
+
+        _stakeDepositTokens(depositTokenAmount);
+        emit Reinvest(totalDeposits(), totalSupply);
+    }
+
+    function _stakeDepositTokens(uint256 _amount) private {
+        require(_amount > 0, "VariableRewardsStrategy::Stake amount too low");
+        _depositToStakingContract(_amount);
+    }
+
+    function _checkReward() internal view returns (Reward[] memory, uint256) {
+        Reward[] memory rewards = _pendingRewards();
+        uint256 estimatedTotalReward = WAVAX.balanceOf(address(this));
+        for (uint256 i = 0; i < rewards.length; i++) {
+            address reward = rewards[i].reward;
+            address swapPair = rewardSwapPairs[rewards[i].reward];
+            uint256 balance = IERC20(reward).balanceOf(address(this));
+            if (reward != address(WAVAX)) {
+                uint256 amount = balance.add(rewards[i].amount);
+                if (amount > 0 && swapPair > address(0)) {
+                    estimatedTotalReward = estimatedTotalReward.add(
+                        DexLibrary.estimateConversionThroughPair(amount, reward, address(WAVAX), IPair(swapPair))
+                    );
+                }
+                rewards[i].amount = amount;
+            } else {
+                estimatedTotalReward = estimatedTotalReward.add(rewards[i].amount);
+            }
+        }
+        return (rewards, estimatedTotalReward);
+    }
+
+    function checkReward() public view override returns (uint256) {
+        (, uint256 estimatedTotalReward) = _checkReward();
+        return estimatedTotalReward;
+    }
+
+    /**
+     * @notice Estimate recoverable balance after withdraw fee
+     * @return deposit tokens after withdraw fee
+     */
+    function estimateDeployedBalance() external view override returns (uint256) {
+        uint256 depositBalance = totalDeposits();
+        uint256 withdrawFee = _calculateWithdrawFee(depositBalance);
+        return depositBalance.sub(withdrawFee);
+    }
+
+    function rescueDeployedFunds(uint256 _minReturnAmountAccepted, bool _disableDeposits) external override onlyOwner {
+        uint256 balanceBefore = depositToken.balanceOf(address(this));
+        _emergencyWithdraw();
+        uint256 balanceAfter = depositToken.balanceOf(address(this));
+        require(
+            balanceAfter.sub(balanceBefore) >= _minReturnAmountAccepted,
+            "VariableRewardsStrategy::Emergency withdraw minimum return amount not reached"
+        );
+        emit Reinvest(totalDeposits(), totalSupply);
+        if (DEPOSITS_ENABLED == true && _disableDeposits == true) {
```

Comment:
we should always disable deposits after emergencyWithdraw. Maintain interface `rescueDeployedFunds(uint256,bool)` but disregard param `_disableDeposits`

Reply:
fixed